src/middlewares/authMiddleware.ts

```diff
@@ -0,0 +1,29 @@
+import 'dotenv/config';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import { IUser } from '../types/userTypes.js';
+import User from '../models/user.js';
+
+const secret: string = process.env.SECRET_KEY || '';
+
+const authMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const authorizationToken = req.headers.authorization;
+
+        if (authorizationToken !== undefined) {
+            const token = authorizationToken.split(' ')[1];
+            const { id } = jwt.verify(token, secret) as JwtPayload;
+
+            if (id) {
+                const user = await User.findById(id);
```

Comment:
Do not load user in middleware. In this case, you will load the user even if you don't need it.
Just store decoded information from token